src/Server.php

```diff
@@ -227,10 +251,10 @@ public function handleRequest(ConnectionInterface $conn, ServerRequestInterface
             $conn->write("HTTP/1.1 100 Continue\r\n\r\n");
         }
 
-        $callback = $this->callback;
+        $middlewareStack = $this->middlewareStack;
         $cancel = null;
-        $promise = new Promise(function ($resolve, $reject) use ($callback, $request, &$cancel) {
-            $cancel = $callback($request);
+        $promise = new Promise\Promise(function ($resolve, $reject) use ($middlewareStack, $request, &$cancel) {
+            $cancel = $middlewareStack->process($request);
             $resolve($cancel);
         });
```

Comment:
Since `MiddlewareStack::process` always returns a promise, this can be simplified to just

```php
$cancel = $middlewareStack->process($request);
```